pkg/netconfig/netconfig.go

```diff
@@ -0,0 +1,339 @@
+		err := IPTablesLoggingChainRule(ipVersion, "tcp", ipSet, defaultNetworkDevice, false, blockIngress)
+		if err != nil {
+			return fmt.Errorf("Error creating tcp logging chain rules for %s, device %s %v\n", ipSet, defaultNetworkDevice, err)
+		}
+	}
+	if err := IPTablesLoggingChainRule(ipVersion, "udp", ipSet, defaultNetworkDevice, true, blockIngress); err != nil {
+		err := IPTablesLoggingChainRule(ipVersion, "udp", ipSet, defaultNetworkDevice, false, blockIngress)
+		if err != nil {
+			return fmt.Errorf("Error creating udp logging chain rules for %s, device %s %v\n", ipSet, defaultNetworkDevice, err)
+		}
+	}
+	return nil
+}
+
+func InitIPSet(ipVersion, ipSetName string) error {
+	if ipVersion == "4" {
+		ipVersion = ""
+	}
+	fmt.Printf("Initially creating ipset with name %s\n", ipSetName)
+	if err := DefaultNetUtilsCommandExecutor.ExecuteIPSetCommand("list", ipSetName); err != nil {
+		err := DefaultNetUtilsCommandExecutor.ExecuteIPSetCommand("create", ipSetName, "hash:net", "family", "inet"+ipVersion, "maxelem", ipSetsMaxLen)
+		if err != nil {
+			return errors.New(fmt.Sprintf("Error creating IP set %s: %v\n", ipSetName, err))
+		}
+	}
+	return nil
+}
+
+func prepareAddrs(content string, trimSuffix bool) []string {
+	res := []string{}
+	lines := strings.Split(content, "\n")
+	for _, line := range lines {
+		if !strings.Contains(line, "[]") {
+			if len(strings.Fields(line)) == 2 {
+				addr := strings.Fields(line)[1]
+				if trimSuffix && strings.HasSuffix(addr, "/32") {
+					addr = strings.TrimSuffix(addr, "/32")
+				}
+				res = append(res, addr)
+			}
+		}
+	}
+	return res
+}
+
+func AddIPListToIPSet(ipSetName string, content string) error {
+	addrs := prepareAddrs(content, false)
+
+	for _, addr := range addrs {
+		err := DefaultNetUtilsCommandExecutor.ExecuteIPSetCommand("-A", ipSetName, addr)
+		if err != nil {
+			fmt.Printf("Error: Failed to add %s to ipset %s: %v\n", addr, ipSetName, err)
+		}
+	}
+	return nil
+}
+
+func UpdateIPSet(ipVersion, ipSetName, egressFilterList, defaultNetworkDevice string, blockIngress bool) error {
+	inetVersion := ""
+	if ipVersion == "6" {
+		inetVersion = "6"
+	}
+	
+	defer func() {
+		fmt.Println("Clean-up")
+		err := DefaultNetUtilsCommandExecutor.ExecuteIPSetCommand("destroy", "tmpIPSet")
+		if err != nil {
+			fmt.Printf("Error cleaning-up ipsets %v\n", err)
+		}
+	}()
+
+	fmt.Printf("Creating temporary ipset with name \"%s\"...\n", "tmpIPSet")
+	err := DefaultNetUtilsCommandExecutor.ExecuteIPSetCommand("create", "tmpIPSet", "hash:net", "family", "inet"+inetVersion, "maxelem", ipSetsMaxLen)
+	if err != nil {
+		return fmt.Errorf("Error creating temporary ipset: %w\n", err)
+	}
+
+	fmt.Printf("Temporary ipset with name \"%s\" created successfully.\n", "tmpIPSet")
+
+	err = AddIPListToIPSet("tmpIPSet", egressFilterList)
+	if err != nil {
+		return fmt.Errorf("Error adding entries to temporary ipset: %w\n", err)
+	}
+	fmt.Println("Added entries into temporary ipset successfully.")
```

Comment:
What do you think about adding some counters? Previously, we could at least see how many entries were created, which can help in understanding whether the list changed. I would prefer to have this here as well. It should be fairly easy to export the data.

Reply:
I added log messages, including total entries, added and deleted entries.